python/pyarrow/io.pxi

```diff
@@ -1337,6 +1340,9 @@ cdef class Buffer(_Weakrefable):
             return pyarrow_wrap_buffer(parent_buf)
 
     def __getitem__(self, key):
+        if (not self.is_cpu):
```

Comment:
FWIW You don't need the brackets here

Reply:
Oh, I am getting confused switching between C++ and Python 🤣